packages/wordpress-playground-block/src/view.tsx

```diff
@@ -7,15 +7,40 @@ function renderPlaygroundPreview() {
 	const playgroundDemo = Array.from(
 		document.getElementsByClassName('wordpress-playground-block')
 	);
-
-	for (const element of playgroundDemo) {
-		const rootElement = element as HTMLDivElement;
+	const urlParams = new URLSearchParams(location.search);
+	if (
+		urlParams.has('playground-full-page') &&
+		urlParams.has('playground-attributes') &&
+		playgroundDemo.length === 1
+	) {
+		const rootElement = playgroundDemo[0] as HTMLDivElement;
 		const root = createRoot(rootElement);
+		const encodedAttributes = urlParams.get(
+			'playground-attributes'
+		) as string;
+		const attributeJson = atob(encodedAttributes);
```

Comment:
Just to make sure – are there two layers of base64 encoding here? So the entire attributes string and each attribute value individually?

Reply:
`atob` and `btoa` won't correctly process UTF-8 characters, e.g.:

```js
> btoa('zażółć gęślą jaźń')
VM326:1 Uncaught DOMException: Failed to execute 'atob' on 'Window': The string to be decoded contains characters outside of the Latin1 range.
```

Instead, use `base64ToString` and `stringToBase64` exported from `base64.ts`:

```js
> stringToBase64( 'zażółć-gęślą-jaźń' );
'emHFvMOzxYLEhy1nxJnFm2zEhS1qYcW6xYQ='
```

Reply:
I went ahead and adjusted that in [4ee7a8c](https://github.com/WordPress/playground-tools/pull/326/commits/4ee7a8c953249d4373571f470f549341e471f28f)

Reply:
> I went ahead and adjusted that in [4ee7a8c](https://github.com/WordPress/playground-tools/pull/326/commits/4ee7a8c953249d4373571f470f549341e471f28f)

Thanks for catching that and making the fix! ❤️